copy/copy.go

```diff
@@ -1426,7 +1530,9 @@ func (c *copier) compressGoroutine(dest *io.PipeWriter, src io.Reader, compressi
 	if err != nil {
 		return
 	}
-	defer compressor.Close()
+	defer func() {
+		err = compressor.Close()
```

Comment:
Typically the error from `CopyBuffer` is much more important, because it probably contains an error reading from the source; it must not be just discarded. If the `Close` error becomes important as well, maybe both need to be reported.

Reply:
thanks for noticing it, I pushed a fixed version